sharktank/sharktank/layers/base.py

```diff
@@ -33,12 +41,82 @@ def _set_recursively_submodules_default_trace_tensor_key_prefix(
         )
 
 
-class BaseLayer(nn.Module):
+def get_model_type_id(model_type: type["BaseLayer"]) -> str:
+    """Get a string representation of the model type."""
+    return f"{model_type.__module__}.{model_type.__name__}"
+
+
+def create_model(config: PathLike | ModelConfig | Mapping[str, Any], /) -> "BaseLayer":
+    """
+    Create model from a configuration.
+    Example
+
+    config.json:
+    ```
+    {
+        "model_type": "MyModel",
+        "mlir_path": "model.mlir",
+        "export_parameters_path": "model.irpa",
+        "iree_module_path": "model.vmfb",
```

Comment:
I presume these paths are relative to the JSON file, or they could be absolute?

edit: I see some notes in another file. If these files reach broader adoption it would be worth centralizing the documentation and linking to that one location.

Reply:
They can be relative to the config file directory or be absolute. There is doc describing this in the `ModelConfig` class.